src/toil_scripts/adam_pipeline/adam_preprocessing.py

```diff
@@ -128,21 +134,40 @@ def remove_file(masterIP, filename, sparkOnToil):
     :type masterIP: MasterAddress
     """
     masterIP = masterIP.actual
+
+    ssh_call = ['ssh', '-o', 'StrictHostKeyChecking=no', masterIP]
+
     if sparkOnToil:
-        try:
-            output = check_output(['ssh',
-                                   '-o', 'StrictHostKeyChecking=no',
-                                   masterIP, 'docker', 'ps'])
-            containerID = next(line.split()[0] for line in output.splitlines() if 'apache-hadoop-master' in line)
-            check_call(['ssh',
-                        '-o', 'StrictHostKeyChecking=no',
-                        masterIP,
-                        'docker', 'exec', containerID,
-                        'hdfs', 'dfs', '-rm', '-r', '/' + filename])
-        except:
-            pass
-    else:
-        log.warning('Cannot remove file %s. Can only remove files when running Spark-on-Toil', filename)
+        output = check_output(ssh_call +  ['docker', 'ps'])
+        containerID = next(line.split()[0] for line in output.splitlines() if 'apache-hadoop-master' in line)
+        ssh_call += ['docker', 'exec', containerID]
+
+    try:
+        check_call(ssh_call + ['hdfs', 'dfs', '-rm', '-r', '/' + filename])
+    except:
+        pass
+
+
+def truncate_file(masterIP, filename, sparkOnToil):
+    """
+    Truncate the given hdfs file to 0 bytes with master at the given IP address
+
+    :type masterIP: MasterAddress
+    """
+    masterIP = masterIP.actual
+
+    ssh_call = ['ssh', '-o', 'StrictHostKeyChecking=no', masterIP]
+
+    if sparkOnToil:
+        output = check_output(ssh_call +  ['docker', 'ps'])
+        containerID = next(line.split()[0] for line in output.splitlines() if 'apache-hadoop-master' in line)
+        ssh_call += ['docker', 'exec', containerID]
+
+    try:
+        check_call(ssh_call + ['hdfs', 'dfs', '-truncate', '-w', '10', '/' + filename])
+    except:
+        pass
+
```

Comment:
Feels like this hunk has nothing to do with mock mode. Should it be committed separately?